src/components/PostDetails.tsx

```diff
@@ -1,117 +1,147 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { client } from '../utils/fetchClient';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const { id, title, body } = post;
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [commentsLoadError, setCommentsLoadError] = useState('');
+  const [commentsDelError, setCommentsDelError] = useState<number[]>([]);
+  const [addNewComment, setAddNewComment] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    if (commentsLoadError) {
+      setCommentsLoadError('');
+    }
+
+    if (comments) {
+      setComments(null);
+    }
+
+    if (addNewComment) {
+      setAddNewComment(false);
+    }
```

Comment:
why do you need to reset this value here, better to create a separate func for this and do it in add func after clicking on submit but not on re rendering component